File: ios-swiftui/library/Sources/PlanetComponents/TspAlert.swift
import SwiftUI

public struct TspAlert: View {
    public enum Variant { case info, success, warning, error }
    let title: String
    let message: String
    let variant: Variant
    let theme: StarPlanetTheme

    public init(title: String, message: String, variant: Variant = .info, theme: StarPlanetTheme = .sky) {
        self.title = title
        self.message = message
        self.variant = variant
        self.theme = theme
    }

    public var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            Text(title).font(.system(size: 15, weight: .bold)).foregroundColor(theme.textPrimary)
            Text(message).font(.system(size: 13)).foregroundColor(theme.textSecondary)
        }
        .padding(14)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(fill)
        .overlay(RoundedRectangle(cornerRadius: 18).stroke(stroke, lineWidth: 1))
        .clipShape(RoundedRectangle(cornerRadius: 18))
    }

    private var fill: Color {
        switch variant {
        case .success: return theme.selectedFill
        case .warning: return theme.activeFill
        case .error: return theme.danger.opacity(0.12)
        case .info: return theme.pageEnd
        }
    }

    private var stroke: Color {
        switch variant {
        case .success: return theme.success
        case .warning: return theme.warning
        case .error: return theme.danger
        case .info: return theme.borderDefault
        }
    }
}

File: flutter/library/lib/src/tsp_alert.dart
import 'package:flutter/material.dart';
import 'star_planet_theme.dart';

enum TspAlertVariant { info, success, warning, error }


class TspAlert extends StatelessWidget {
  const TspAlert({super.key, required this.title, required this.message, this.variant = TspAlertVariant.info, this.theme = StarPlanetTheme.sky});
  final String title;
  final String message;
  final TspAlertVariant variant;
  final StarPlanetTheme theme;

  @override
  Widget build(BuildContext context) {
    final fill = switch (variant) {
      TspAlertVariant.success => theme.selectedFill,
      TspAlertVariant.warning => theme.activeFill,
      TspAlertVariant.error => theme.danger.withValues(alpha: .12),
      TspAlertVariant.info => theme.pageEnd,
    };
    final stroke = switch (variant) {
      TspAlertVariant.success => theme.success,
      TspAlertVariant.warning => theme.warning,
      TspAlertVariant.error => theme.danger,
      TspAlertVariant.info => theme.borderDefault,
    };
    return Container(
      width: double.infinity,
      padding: const EdgeInsets.all(14),
      decoration: BoxDecoration(color: fill, borderRadius: BorderRadius.circular(18), border: Border.all(color: stroke)),
      child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
        Text(title, style: TextStyle(color: theme.textPrimary, fontWeight: FontWeight.w800)),
        const SizedBox(height: 6),
        Text(message, style: TextStyle(color: theme.textSecondary, fontSize: 13)),
      ]),
    );
  }
}
